src/sentry/api/endpoints/group_notes.py

```diff
@@ -32,33 +32,48 @@ def get(self, request, group):
 
     def post(self, request, group):
         serializer = NoteSerializer(data=request.DATA)
+        serializer2 = MentionSerializer(data=request.DATA)
+
         if not serializer.is_valid():
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-        data = dict(serializer.object)
+        if not serializer2.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        note_data = dict(serializer.object)
+
+        GroupSubscription.objects.subscribe(
+            group=group,
+            user=request.user,
+            reason=GroupSubscriptionReason.comment,
+        )
+        mentions_data = dict(serializer2.object)
+
+        if mentions_data['mentions']:
+            for item in mentions_data['mentions']:
+                user = User.objects.get(id=item)
+                GroupSubscription.objects.subscribe(
+                    group=group,
+                    user=user,
+                    reason=GroupSubscriptionReason.mentioned,
+                )
 
```

Comment:
so a few things need to happen here:
* rather than querying for each user individually, you should use `User.objects.filter(id__in=mentions_data['mentions'])` because then it will only take one query no matter the number of users
* you should also return an error message/400 status when a user isn't found
* i'm not too familiar with the subscription code, but I think you need to also check to make sure that the mentioned user is a member of the organization / has access to that project, etc